### DISCUSSION TITLE
Additional Fill conditions

### DISCUSSION THREAD

**Discussion by MRBBATES1** [Mojang/minecraft-editor#30] [Feedback]
At the moment when using the fill option, will just fill in the area selected, however if you wish to only fill i air blocks with a different block type I cant figure out how to do this without having to make a number of micro selections.

it would be nice to be able to fill air only option leaving existing blocks / mobs in place or perhaps if you want to expand that have it as a replace feature instead.

**Comment by Landy005** [comment]
Basically how fast async world edit handles //set or //replace with flags and gmask to not include or to include certainblocks, entities, biomes. Paste or copy without air or water entities and biomes would also be a great addition to this!

**Reply by CaptainCooky** [reply]
Agreed, being able to mask blocks is essential. Otherwise, creating unorthodox shapes, would be too time consuming, and better of being done manually.

**Comment by minecraftchloe** [comment]
Yes! We 100% agree with this idea!! And also, with the Editor API, building extensions that have these additional conditions will be possible.
